packages/@aws-cdk/aws-codebuild/lib/linux-gpu-build-image.ts

```diff
@@ -109,37 +107,34 @@ export class LinuxGpuBuildImage implements IBindableBuildImage {
 
   public readonly type = 'LINUX_GPU_CONTAINER';
   public readonly defaultComputeType = ComputeType.LARGE;
-  public readonly imageId: string;
   public readonly imagePullPrincipalType?: ImagePullPrincipalType = ImagePullPrincipalType.SERVICE_ROLE;
+  public readonly imageId: string;
 
-  private readonly accountExpression: string;
+  private _imageId?: string;
 
-  private constructor(private readonly repositoryName: string, tag: string, private readonly account: string | undefined) {
-    this.accountExpression = account ?? core.Fn.findInMap(mappingName, core.Aws.REGION, 'repositoryAccount');
-    this.imageId = `${this.accountExpression}.dkr.ecr.${core.Aws.REGION}.${core.Aws.URL_SUFFIX}/${repositoryName}:${tag}`;
+  private constructor(private readonly repositoryName: string, private readonly tag: string, private readonly account: string | undefined) {
+    this.imageId = core.Lazy.string({
```

Comment:
This is a potentially breaking change, no?

```ts
const myImage = codebuild.LinuxGpuBuildImage.DLC_TENSORFLOW_1_14_0;
createCBJob(myImage);

// ...
function createCBJob(myImage: codebuild.IBindableBuildImage) {
  const jobId = myImage.imageId.includes('pytorch-training') ? 'pytorchjob' : 'tensorflowjob';
  new codebuild.Project(this, jobId, { ... });
}
```

Obviously, a fabricated example, but can we really convert a string from a concrete to Lazy, tokenized value in a stable module and guarantee we won't break customers?

Reply:
**Sigh**. I suppose not, and it's a good catch. But based on those kinds of arguments I don't think we can't make any changes at all because anybody anywhere might have taken a dependency on anything.

I want to say how big is the chance people have actually done that... and I think not too big?